misc/python/materialize/data_ingest/data_type.py

```diff
@@ -0,0 +1,187 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import random
+import string
+from enum import Enum
+from typing import Any
+
+
+class RecordSize(Enum):
+    TINY = 1
+    SMALL = 2
+    MEDIUM = 3
+    LARGE = 4
+
+
+class Backend(Enum):
+    AVRO = 1
+    POSTGRES = 2
+
+
+class DataType:
+    """As supported by Avro: https://avro.apache.org/docs/1.11.1/specification/_print/"""
+
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        raise NotImplementedError
+
+    @staticmethod
+    def num_value(num: int) -> Any:
```

Comment:
This method seems to be for numeric types only. That means that it does not make sense for `BooleanType`, where it will raise an exception. Would it make sense to introduce a `NumericDataType` class in the hierarchy?